Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -235,14 +240,13 @@ private void AssignPendingVectors()
         }
 
         /// <inheritdoc />
-        public bool IsDownloading(uint256 hash)
+        public void CheckBlockStatus(uint256 hash, out bool IsDownloading, out bool IsReady)
```

Comment:
I suggest instead of using out param to return the new anonymous tuple result
```
 public (bool IsDownloading, bool IsReady) CheckBlockStatus(uint256 hash)
```